source/common/upstream/load_balancer_impl.cc

```diff
@@ -87,18 +87,23 @@ void LoadBalancerBase::recalculatePerPriorityState(uint32_t priority,
   }
 
   size_t total_load = 100;
+  int32_t first_healthy_priority = -1;
   for (size_t i = 0; i < per_priority_health.size(); ++i) {
-    // Now assign as much load as possible to the high priority levels and cease assigning load when
-    // total_load runs out.
+    if (first_healthy_priority < 0 && per_priority_health[i] > 0) {
+      first_healthy_priority = i;
+    }
+    // Now assign as much load as possible to the high priority levels and cease assigning load
+    // when total_load runs out.
     per_priority_load[i] =
         std::min<uint32_t>(total_load, per_priority_health[i] * 100 / total_health);
     total_load -= per_priority_load[i];
   }
 
   if (total_load != 0) {
-    // Account for rounding errors.
+    ASSERT(first_healthy_priority != -1);
+    // Account for rounding errors by assigning it to the first healthy priority.
```

Comment:
Oh nice improvement.  Mind adding both a load balancer test and adding this to the commit notes?

Reply:
Huh this was actually added in https://github.com/envoyproxy/envoy/pull/4533, not sure how this ended up in this diff. I'll see if I can fix it

Reply:
Haha, the joys of git :-P

Reply:
I merged in a bunch of unrelated commits and the diff went away /shrug